ArjunaJTA/jta/classes/com/arjuna/ats/internal/jta/recovery/arjunacore/JTATransactionLogXAResourceOrphanFilter.java

```diff
@@ -101,32 +103,24 @@ private boolean transactionLog(Xid xid)
 
         if (!u.equals(Uid.nullUid()))
         {
-            try
-            {
+            if (jtaLogger.logger.isDebugEnabled()) {
+                jtaLogger.logger.debug("Looking for " + u + " and " + transactionType);
+            }
 
+            if (containsCommitMarkableResourceRecord(u) ||
+                    recoveryStore.currentState(u, transactionType) != StateStatus.OS_UNKNOWN)
+            {
                 if (jtaLogger.logger.isDebugEnabled()) {
-                    jtaLogger.logger.debug("Looking for " + u + " and " + transactionType);
+                    jtaLogger.logger.debug("Found record for " + theXid);
                 }
 
-                if (containsCommitMarkableResourceRecord(u) ||
-                        recoveryStore.currentState(u, transactionType) != StateStatus.OS_UNKNOWN)
-                {
-                    if (jtaLogger.logger.isDebugEnabled()) {
-                        jtaLogger.logger.debug("Found record for " + theXid);
-                    }
-
-                    return true;
-                }
-                else
-                {
-                    if (jtaLogger.logger.isDebugEnabled()) {
-                        jtaLogger.logger.debug("No record found for " + theXid);
-                    }
-                }
+                return true;
             }
-            catch (Exception ex)
+            else
             {
-                ex.printStackTrace();
+                if (jtaLogger.logger.isDebugEnabled()) {
+                    jtaLogger.logger.debug("No record found for " + theXid);
+                }
```

Comment:
Its a fair point. Because my test throws ISE the exception is actually caught (and logged) in XARM: https://github.com/jbosstm/narayana/blob/master/ArjunaJTA/jta/classes/com/arjuna/ats/internal/jta/recovery/arjunacore/XARecoveryModule.java#L738

I will add a line of debug incase OSE or IOE are thrown though